### PR TITLE
[Issue 892] Add API_AUTH_TOKEN environment variable

### LINKED ISSUE #892: Configure authn key
### Acceptance Criteria

- [ ] Configure API_AUTH_TOKEN in the appropriate environments - test and prod

### PR DESCRIPTION
**Pull request by chouinar**
## Summary
Fixes #892

## Changes proposed
Set the `API_AUTH_TOKEN` in terraform

## Context for reviewers
NOTE: As I want to verify the naming I did for the parameter store variable, I haven't yet added a prod key to parameter store - will do before merging.

While we may adjust how our authentication works in the future, we want to be able to at least use our API in the dev environment. Right now there is not an auth token setup, I added a new one in parameter store for dev (generated something random with 1password)

## Additional information
Let me know if there's anything off about how I set this up / tested it. Been a while since I did any Terraform, and that was setup a bit different.

I can tell it is fetching from parameter store as before I set it up, it complained when doing `terraform plan`
Screenshot 2023-12-19 at 12 16 43 PM

After adding it to parameter store, I now see:
Screenshot 2023-12-19 at 4 51 02 PM

Which seems right from what I recall about Terraform?

### CODE REVIEW

File: infra/modules/service/variables.tf [hunk 1]
```diff
@@ -97,3 +97,9 @@ variable "min_capacity" {
   type        = number
   default     = 2
 }
+
+variable "api_auth_token" {
```
Comment (coilysiren) [inline infra/modules/service/variables.tf:101]: ⭐⭐  &nbsp; _(optional but seriously consider)_

```suggestion
variable "api_auth_token" {
  sensitive   = true
```

**Review by coilysiren — COMMENTED** [review]
The PR looks good, but the unit test failures seem related. We probably want to look into what's going on there!

**Comment by coilysiren** [comment]
> Which seems right from what I recall about Terraform?

Yes the task definition needs to be replaced when you update an env var, that's an AWS limitation

**Comment by coilysiren** [comment]
The test failures might just be because you haven't added the parameter yet? Interested in what's going on there, I haven't used terraform test frameworks before!

**Comment by chouinar** [comment]
> The test failures might just be because you haven't added the parameter yet? Interested in what's going on there, I haven't used terraform test frameworks before!

I reran it so I could get more recent logs on AWS and it passed. So..? It failed before waiting for the service to be available so maybe there was just a network blip?

**Review by acouch — APPROVED** [review]

**Comment by acouch** [comment]
FYI we may need to remove the auth token for the `v0` release due to our not providing any auth for the initial release.

**Comment by lucasmbrown-usds** [comment]
> FYI we may need to remove the auth token for the `v0` release due to our not providing any auth for the initial release.

Can we use a feature flag to enable/disable the auth token, rather than removing the code?

**Comment by chouinar** [comment]
> > FYI we may need to remove the auth token for the `v0` release due to our not providing any auth for the initial release.
> 
> Can we use a feature flag to enable/disable the auth token, rather than removing the code?

1. Why wouldn't we want this basic auth? I was actually going to suggest we add something more to be able to distinguish our users in addition to this.
2. We could use a feature flag if needed, although would want to discuss exactly how we want to add that (a few options).

**Pull request merged**